File: Java/Array_101/3_Deleting/RemoveElement.java
public class RemoveElement {
    class Solution {
        public int removeElement(int[] nums, int val) {
            int n = nums.length;
            int ret = n;
            int t;
            for (int i = 0; i < n; i++) {
                if (nums[i] == val) {
                    for (int j = n - 1; j > i; j--) {
                        nums[j - 1] = nums[j];
                    }
                    i++;
                    ret--;
                }
            }
            return ret;
        }
    }
}

File: Python/Array_101/3_Deleting/remove_element.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        n = len(nums)
        ret = n
        firstIdx = -1
        for i in range(n):
            if nums[i] == val:
                ret -= 1
                if firstIdx == -1:
                    firstIdx = i
        if firstIdx != -1:
            movIdx = firstIdx
            for i in range(firstIdx, ret):
                for j in range(movIdx+1, n):
                    if nums[j] != val:
                        nums[i] = nums[j]
                        movIdx = j
                        break
            
        return ret